discord/widget.py

```diff
@@ -0,0 +1,217 @@
+        """:class:`DefaultAvatar`: The same operation as :meth:`User.default_avatar`."""
+        return DefaultAvatar(int(self.discriminator) % len(DefaultAvatar))
+
+    @property
+    def default_avatar_url(self):
+        """Returns a URL for a user's default avatar."""
+        return 'https://cdn.discordapp.com/embed/avatars/{}.png'.format(self.default_avatar.value)
+
+class Widget:
+    """Represents a :class:`Guild` widget.
+
+    This model is always given, as long as it is enabled.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The guild's ID.
+    name: :class:`str`
+        The guild's name.
+    channels: List[:class:`VoiceChannel`]
+        The accessible voice channels in the guild.
+    members: List[:class:`Member`]
+        The online members in the server. Note: offline
+        members do not appear in the widget.
+    invite: Optional[:class:`Invite`]
+        The invite set for the widget.
+    """
+    __slots__ = ('_state', 'channels', 'invite', 'id', 'members', 'name')
+
+    def __init__(self, *, data, invite):
+        self.invite = invite
+        self.name = data.get('name')
+        self.id = int(data.get('id'))
+
+        self.channels = []
+        for channel in data.get('channels'):
+            _id = int(channel['id'])
+            self.channels.append(WidgetChannel(id=_id, name=channel['name'], position=channel['position']))
+
+        self.members = []
+        for member in data.get('members'):
+            _id = int(member.get('id'))
+            bot = bool(member.get('bot'))
+            status = try_enum(Status, member.get('status'))
+            deafened = bool(member.get('deaf')) or bool(member.get('self_deaf'))
+            muted = bool(member.get('mute')) or bool(member.get('self_mute'))
+            suppress = bool(member.get('mute'))
+
+            game = None
+            if member.get('game'):
+                game = Game(**member.get('game'))
+
+            connected_channel = None
+            if member.get('channel_id'):
+                connected_channel = get(self.channels, id=int(member.get('channel_id')))
+
+            self.members.append(WidgetMember(id=_id,
+                                             bot=bot,
+                                             username=member.get('username'),
+                                             discriminator=member.get('discriminator'),
+                                             status=status,
+                                             avatar=member.get('avatar'),
+                                             nick=member.get('nick'),
+                                             game=game,
+                                             deafened=deafened,
+                                             muted=muted,
+                                             suppress=suppress,
+                                             connected_channel=connected_channel))
+
+    def __str__(self):
+        return self.json_url
```

Comment:
This needs to be documented. Usually there is an `.. container:: operations` table that documents the python ops like `str`, `hash`, `==`, etc you can do on a type.